share/functions/__fish_complete_wvdial_peers.fish

```diff
@@ -15,15 +15,15 @@ function __fish_complete_wvdial_peers --description 'Complete wvdial peers' --ar
 		case -C --config
 			set store_next true
 		case '--config=*'
-			set cfgfiles (echo $opt | sed 's/--config=//')
+			set cfgfiles (echo $opt | string replace '--config=' '')
 		end
 	end
 
 	for file in $cfgfiles
         if test -f $file
-            cat $file | grep '\[Dialer' | sed 's/\[Dialer \(.\+\)\]/\1/'
+            string match -a '\*\[Dialer\*' < $file | sed 's/\[Dialer \(.\+\)\]/\1/'
```

Comment:
Sure you want to have the `*` escaped? In glob mode that now matches a literal `*` character, so only the line `*\[Dialer\*` will be matched here. (And the "-a" is probably unnecessary). How about `string match '*[Dialer]*' - the "[" don't have special meaning in glob mode.